### ISSUE TITLE
Wrong handling of Windows paths (SC1001)

### ISSUE THREAD

**Issue by okainov** [hadolint/hadolint#643]
- [x] This is a bug report
- [ ] This is a feature request
- [ ] I searched existing issues before opening this one

As #371 was implemented, now we can use hadolint for Windows Dockerfiles. But seems like it doesn't like Windows paths.

Dockerfile:
```
FROM mcr.microsoft.com/dotnet/framework/sdk:4.8

RUN C:\BuildTools\qqq.exe
```

### Expected behavior

No lint errors

### Actual behavior

```
docker run --rm -i hadolint/hadolint < Dockerfile
-:3 SC1001 info: This \q will be a regular 'q' in this context.
```

### Steps to reproduce the behavior

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**

```bash
Haskell Dockerfile Linter v2.4.1-0-g4b95e40
```

**Dockerfile** (if relevant)

```Dockerfile
(paste content of your Dockerfile)
```

**Additional environment details (OS, stack version, etc.)**
